lua/nvim-tree/view.lua

```diff
@@ -202,12 +202,23 @@ function M.focus(winnr, open_if_closed)
   a.nvim_set_current_win(wnr)
 end
 
+local function handle_width()
+  if type(M.View.width) == "string" then
+    if string.sub(M.View.width, -1) == "%" then
+      local width_as_number = math.floor(string.sub(M.View.width, 0, -2))
```

Comment:
This substring is not gonna work for single digit, or triple digit percentage. Please use a pattern for this instead. And don't floor the number here. Allow users to use decimal percentage. You are flooring the final result anyway.

```lua
if vim.endswith(M.View.width, "%") then
  local width_as_number = tonumber(M.View.width:match('[^%d]*([%d%.]+)%%') or 30)
```

Reply:
Awesome pattern here!
Would this suffice?
```lua
local function get_width()
  if vim.endswith(M.View.width, "%") then
    local width_as_number = tonumber(M.View.width:match('[^%d]*([%d%.]+)%%'))
    local percent_as_decimal = width_as_number / 100
    return math.floor(vim.o.columns * percent_as_decimal)
  end
  return M.View.width
end
```

Reply:
Yes, that's great. But you still need the type check, because `vim.endswith` expects a string and will throw an error otherwise.